lib/cfa/augeas_parser.rb

```diff
@@ -47,29 +65,49 @@ def augeas_name
     end
   end
 
-  # Represent parsed augeas config tree with user friendly methods
+  # Represents a parsed Augeas config tree with user friendly methods
   class AugeasTree
-    # low level access to augeas structure
+    # Low level access to Augeas structure
+    #
+    # An ordered mapping, represented by an Array of Hashes
+    # with the keys :key and :value.
+    #
+    # @see AugeasElement
+    #
+    # @return [Array<Hash{Symbol => String, AugeasTree}>]
     attr_reader :data
 
     def initialize
       @data = []
     end
 
+    # @return [AugeasCollection] collection for *key*
     def collection(key)
       AugeasCollection.new(self, key)
     end
 
+    # @param key [String]
     def delete(key)
       @data.reject! { |entry| entry[:key] == key }
     end
 
+    # Add a new *key* with a *value*.
+
+    # By default an AppendPlacer is used which produces duplicate keys
+    # but ReplacePlacer can be used to replace the *first* duplicate.
+    # @todo what if we want to replace all duplicates?
```

Comment:
this method is not suitable for it as its purpose is to add single key with value.